src/util.cpp

```diff
@@ -105,9 +105,84 @@ std::string to_internal(const std::string &);
 
 using namespace std;
 
+namespace Logging {
+//Globals defined here because link fails if in globals.cpp.
+//Keep at top of file so init first:
+	uint64_t categoriesEnabled = 0; //64 bit log id mask.
+	static map<uint64_t, std::string> logLabelMap = LOGLABELMAP; //Lookup log label from log id.
+
+
+uint64_t LogFindCategory(const std::string label)
+{
+	for (auto& x: logLabelMap)
+	{
+		if((std::string)x.second == label)
+			return (uint64_t)x.first;
+	}
+	return NONE;
+}
+
+std::string LogGetLabel(uint64_t category)
+{
+	string label = "none";
+	if(logLabelMap.count(category)!=0)
+		label = logLabelMap[category];
+
+	return label;
+}
+
+int LogWriteStr(const std::string &str)
+{
+	return LogPrintStr(str);
+}
+
+std::string LogGetAllString()
+{
+	string ret="";
+	for (auto& x: logLabelMap)
+	{
+		if(x.first==ALL||x.first==NONE)
+			continue;
+
+		ret+=(std::string)x.second;
+		if(LogAcceptCategory(x.first))
+			ret+= " on";
+		ret+="\n";
+	}
+
+	return ret;
+}
+
+void LogInit()
+{
+	string category = "";
+	uint64_t catg=Logging::NONE;
+	const vector<string>& categories = mapMultiArgs["-debug"];
+
+	for (string const& cat : categories)
+		{
+			category = boost::algorithm::to_upper_copy(cat);
+			catg = LogFindCategory(category);
+
+			if(catg==NONE) //not a valid category
+				continue;
+
+			if(catg==ALL)
+			{
+				LogTurnOnAll(true);
+				break;
+			}
```

Comment:
It looks like the ALL category will work with the LogTurnOnCategory function.  So the above check is not necessary.  (Even if LogTurnOnCategory(ALL, true) didn't work, I think that there is a strong argument for handling ALL inside the function since it puts the complexity down into the Log subsystem rather than in the app code.

Reply:
I'll change it